sklearn/utils/fixes.py

```diff
@@ -84,6 +86,20 @@ def _object_dtype_isnan(X):
     return X != X
 
 
+#TODO:  remove this when minimum version of scipy >= 1.9.0
+def _yeojohnson_lambda( _neg_log_likelihood, x):
+
+    x = x[~np.isnan(x)]
```

Comment:
Should this line move or not ? (check wrt scipy implementation to get the answer)